contracts/ronin/Maintenance.sol

```diff
@@ -85,26 +85,42 @@ contract Maintenance is IMaintenance, HasValidatorContract, Initializable {
   ) external override {
     IRoninValidatorSet _validator = _validatorContract;
 
-    require(_validator.isBlockProducer(_consensusAddr), "Maintenance: consensus address must be a block producer");
-    require(
-      _validator.isCandidateAdmin(_consensusAddr, msg.sender),
-      "Maintenance: method caller must be a candidate admin"
-    );
-    require(!checkScheduled(_consensusAddr), "Maintenance: already scheduled");
-    require(checkCooldownEnds(_consensusAddr), "Maintainance: cooldown time not end");
-    require(totalSchedules() < maxSchedules, "Maintenance: exceeds total of schedules");
-    require(
-      _startedAtBlock.inRange(block.number + minOffsetToStartSchedule, block.number + maxOffsetToStartSchedule),
-      "Maintenance: start block is out of offset"
-    );
-    require(_startedAtBlock < _endedAtBlock, "Maintenance: start block must be less than end block");
+    if (!(_validator.isBlockProducer(_consensusAddr) || _validator.isCandidateAdmin(_consensusAddr, msg.sender))) {
+      revert ErrUnauthorized(msg.sig);
+    }
```

Comment:
Should not merge the errors, and throw clearly explained ones, for easier debug in DApp and better UX

Reply:
splitted.